lib/libc/newlib/libc-hooks.c

```diff
@@ -104,8 +104,13 @@ static int malloc_prepare(const struct device *unused)
 
 #ifdef USE_MALLOC_PREPARE
 #ifdef CONFIG_MMU
+	/* Need extra for the guard pages (before and after) which we
```

Comment:
Is this always true? Or does this depend on some other configuration which actually places guard pages around allocations?